### PR TITLE
improve karmadactl init

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

make the code more readable

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by carlory** [comment]
/hold

waiting for #2191 to be merged

**Comment by prodanlabs** [comment]
> waiting for #2191 to be merged

Thank @carlory ,  #2191 has been merged.
/hold cancel

**Comment by carlory** [comment]
@prodanlabs Unfortunately, the e2e testing was failed. 

the `lgtm` label is removed, could you review it again ?

**Review by lonelyCZ — COMMENTED** [review]
Wow, I just wanted to improve this code too, thank you have finished it.

File: pkg/karmadactl/cmdinit/kubernetes/options.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	StorageClassesName                 string
+	KarmadaDataPath                    string
+	CRDs                               string
+	ExternalIP                         string
+	ExternalDNS                        string
+	KubeClientSet                      *kubernetes.Clientset
+	CertAndKeyFileData                 map[string][]byte
+	RestConfig                         *rest.Config
+	KarmadaAPIServerIP                 []net.IP
+}
```
[the remark this answers is not in the document]
Comment (lonelyCZ) [reply pkg/karmadactl/cmdinit/kubernetes/options.go:82]: Blank lines by catagory seem clearer.

**Comment by carlory** [comment]
@lonelyCZ @prodanlabs updated

**Comment by lonelyCZ** [comment]
I will review ASAP.

/assign

**Comment by lonelyCZ** [comment]
Hi @carlory , please rebase this branch and resolve conflicts.

I just tested it that worked fine!

**Comment by carlory** [comment]
Please take a look @RainbowMango @prodanlabs

**Comment by RainbowMango** [comment]
I can see this PR include two changes:
1. rename package `pkg/karmadactl/cmdinit/options` to `pkg/karmadactl/cmdinit/constants`
2. move all options to `pkg/karmadactl/cmdinit/kubernetes/options.go`

My questions is this is the options for `pkg/karmadactl/cmdinit`, why put it under `kubernetes`?

**Pull request closed without merging**